drakcore/drakcore/frontend/src/UploadSample.js

```diff
@@ -22,20 +24,62 @@ class UploadSample extends Component {
   async onSubmit(event) {
     event.preventDefault();
     if (this.state.file === null) return;
-    let response = await api.uploadSample(this.state.file);
+    let response = await api.uploadSample(this.state.file, this.state.timeout);
     this.props.history.push(`/progress/${response.data.task_uid}`);
   }
 
+  onInput(event) {
+    this.setState({ timeout: event.target.value });
+  }
+
   render() {
     return (
       <div className="App container-fluid">
         <div className="page-title-box">
           <h4 className="page-title">Upload sample</h4>
         </div>
 
-        <form onSubmit={this.onSubmit}>
-          <input type="file" name="file" onChange={this.onFileChange} />
-          <button type="submit">Upload</button>
+        <form onSubmit={this.onSubmit} className="form-horizontal col-lg-5">
+          <div className="form-group">
+            <div className="custom-file">
+              <label className="custom-file-label" htmlFor="sampleFile">
+                {this.state.file ? this.state.file.name : "Choose file"}
+              </label>
+              <input
+                type="file"
+                name="file"
+                id="sampleFile"
+                className="custom-file-input"
+                onChange={this.onFileChange}
+                required
+              />
+            </div>
+          </div>
+          <div className="form-group">
+            <input
+              onInput={this.onInput}
+              className="custom-range col-10"
+              id="example-range"
+              type="range"
+              name="range"
+              min={60}
+              max={10 * 60}
+              step={60}
+              defaultValue={this.state.timeout}
+            />
```

Comment:
Is it possible to enter the timeout by hand or do we only have this gauge/range/sth?

Reply:
At first, I wanted to add a field to type in the value, however from the UX perspective it seems like a bad idea (especially if we're talking about 1 second resolution).
Are there any usecases when we'd like to have this much control over the analysis time?

Reply:
nvm, lgtm